### PR TITLE
[fix](MTMV) Support star expressions in select list

### LINKED ISSUE #20353: [Bug] create mv as select * failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

CREATE MATERIALIZED VIEW mv1 BUILD DEFERRED REFRESH COMPLETE START WITH "2023-05-16 21:33:00" NEXT 1 DAY 
DISTRIBUTED BY HASH(k1) 
BUCKETS 2 
PROPERTIES (
    "replication_num" = "1"
)
as select * from user1; 

image

### What You Expected?

normal

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #20353 

support create mv support star

adds error prompt for unsupported expr 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMultiTableMaterializedViewStmt.java [hunk 1]
```diff
@@ -117,11 +119,40 @@ private void analyzeSelectClause(SelectStmt selectStmt) throws AnalysisException
...
+                if (!mvColumnPattern.match(functionCallExpr)) {
+                    throw new AnalysisException(
+                            "The function " + functionName + " must match pattern:" + mvColumnPattern);
+                }
+                if (StringUtils.isEmpty(alias)) {
+                    throw new AnalysisException("Function expr: " + functionName + " must have a alias name for MTMV.");
+                }
+            } else {
+                throw new AnalysisException(
+                        "Materialized view does not support this expr:" + itemExpr.toSqlImpl());
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMultiTableMaterializedViewStmt.java:148]: The function `analyzeSelectClause` is too large. Please try to use some small functions to organize the implementation.

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Comment by adonis0147** [comment]
Hi @zddr , this pr should add some tests. Please refer to `MultiTableMaterializedViewTest`.

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
> Hi @zddr , this pr should add some tests. Please refer to `MultiTableMaterializedViewTest`.

done

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**